app/src/main/java/org/oppia/app/customview/inputInteractionView/FractionInputInteractionView.kt

```diff
@@ -0,0 +1,42 @@
+package org.oppia.app.customview.inputInteractionView
+
+import android.content.Context
+import android.text.InputFilter
+import android.text.method.DigitsKeyListener
+import android.widget.EditText
+import org.oppia.app.R
+
+/** The customclass for [FractionInputInteractionView]. */
+class FractionInputInteractionView : EditText {
+
+  private var mContext: Context
+  private var placeholder: String
+  private var rows: Int
+
+  constructor(context: Context, placeholder: String, rows: Int) : super(context) {
+    this.mContext = context
+    this.placeholder = placeholder
+    this.rows = rows
+    init()
+  }
+
+  //update default attributes of TextInputInteractionView here
+  fun init() {
```

Comment:
Kotlin has a concept of Primary and Secondary Constructor and `init` is a type of secondary constructor and its not good to give a function name `init` because of secondary constructor name. Also, I think you can use the concept of primary and secondary constructor here for clearer code.

Reference: https://blog.mindorks.com/primary-and-secondary-constructors-in-kotlin

Reply:
resolved